x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_indices/search_indices.tsx

```diff
@@ -121,11 +139,13 @@ export const SearchIndices: React.FC = () => {
       }),
       actions: [
         {
-          render: ({ indexSlug }: SearchIndex) => (
+          render: ({ name }: SearchIndex) => (
             <EuiButtonIconTo
               iconType="eye"
               data-test-subj="view-search-index-button"
-              to={generatePath(SEARCH_INDEX_OVERVIEW_PATH, { indexSlug })}
+              to={generatePath(SEARCH_INDEX_OVERVIEW_PATH, {
+                indexSlug: encodeURIComponent(name),
```

Comment:
question: I've been using the encodeURI function instead of encodeURIComponent. Any idea which one is preferred?

Reply:
AFAIK: `encodeURI` will not encode everything but only certain characters, `encodeURIComponent` on the other hand will encode everything. 
As this was a part to be in a URI I used component, preference-wise I also don't have an idea :D

Reply:
cool cool cool!